src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -176,15 +179,15 @@ public void removeTaskId(String taskId) {
   }
 
   public boolean journalNodeRunningOnSlave(String hostname) {
-    return getJournalNodes().keySet().contains(hostname);
+    return getJournalNodes().containsKey(hostname);
   }
 
   public boolean nameNodeRunningOnSlave(String hostname) {
-    return getNameNodes().keySet().contains(hostname);
+    return getNameNodes().containsKey(hostname);
   }
 
   public boolean dataNodeRunningOnSlave(String hostname) {
-    return getDataNodes().keySet().contains(hostname);
+    return getDataNodes().containsKey(hostname);
```

Comment:
I modified these because it seemed like an oversight to derive the keyset first. But let me know if there was some other reasoning there.